tbtadm/controller.cpp

```diff
@@ -653,6 +655,188 @@ catch (...)
     m_err << "Unknown exception\n";
 }
 
+void tbtadm::Controller::addToBootACL(const fs::path& dir, std::string uuid)
+{
+    auto fname = dir.filename().string();
+    auto num = fname.substr(0, fname.find('-'));
+    auto boot_acl_path = sysfsDevicesPath / (domain + num) / boot_acl;
+
+    if (!fs::exists(boot_acl_path))
+    {
+        /* Cannot find boot_acl */
+        return;
+    }
+
+    /* Read boot_acl */
+    std::stringstream stream(readAndTrim(boot_acl_path));
+    /* Check maximum allowed boot_acl string size */
+    if (stream.str().size() > 36 * 16 + 1 * 15)
+    {
+        m_out << "boot_acl contains too much characters\n";
+        return;
+    }
+
+    std::queue<std::string> acl_queue;
+
+    while(stream.good())
+    {
+        std::string substring;
+
+        getline(stream, substring, ',');
+	/* this moves also uuid to the end of the queue */
+        if (substring.size() == 36 && substring != uuid)
```

Comment:
What you really care about is that `!substring.empty()`, isn't it?
If so, I think it makes the code clearer and cleaner.

Reply:
Do you mean changing `substring.size() == 36` to `!substring.empty()`? I also check that entry is 36 characters long, I can change it to the proposed way.

Reply:
OK, I see the point of making sure the parsed uuid is 36 chars long, but then, if it isn't 36 nor 0, this should stop as something is wrong (maybe the uuid format has changed?). Isn't it?